syntax = "proto3";

import "logika/resources/enums.proto";

package logika.resources;

message X6TagList
{
    repeated X6Tag list = 1;
}

message X6Tag
{
    string device = 1;
    string channel = 2;
    int32 ordinal = 3;
    optional int32 index = 4;
    X6TagTypeEnum type = 5;
    optional TagKindEnum kind = 6;
    bool basic = 7;
    optional DataTypeEnum dataType = 8;
    int32 updateRate = 9;
    optional string name = 10;
    string description = 11;
    optional VarTypeEnum varType = 12;
    optional int32 count = 13;
    optional string range = 14;
    optional string descriptionEx = 15;
}
